### ISSUE TITLE
Config: Remove code handling CONFIG_VERSION <= 4

### ISSUE THREAD

**Issue by buhtz** [bit-team/backintime#1497]
I did some research in the commit history and try to summarize this here. It is a long term issue and there is no hurry. It is a side benefit in context of DebianBug [#1043046](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1043046) where I needed to investigate the differences between our config versions and if the reported issue may have been caused by a problem in the code handling the conversion between the different config versions.

So I just don't want to lose the information I found and invest it right into improving BIT.

- `config.py::Config` does have a class attribute `CONFIG_VERSION` with value `6`.
- The `CONFIG_VERSION` (with value `2`) was introduced near BIT version [0.9.26](https://github.com/bit-team/backintime/commit/fcd527530e96911add97376e4bfe7fb391b52b39) (or slightly earlier). This was 14 years ago.
- `CONFIG_VERSION = 5` was sent somewhere between BIT version [0.9.26](https://github.com/bit-team/backintime/blob/v0.9.26/common/config.py) (14 years old) and [1.0.0](https://github.com/bit-team/backintime/blob/v1.0.0/common/config.py) (13 years old). Between this two versions, which are direct neighbors the config version jumped from `2` to `5`.

Based on that information and the age of that versions I would say that `CONFIG_VERSION <= 4` is irrelevant today for us and the conversion code can be removed.

Why? It is just a clean up step but it prepares the re-design of the `Config` class and all its related components.

**Comment by Germar** [comment]
I only increased `CONFIG_VERSION` if I changed a config variable which is already in use to a different behavior. But when I added new config variable or added new options to already existing variables it was not necessary to increase because the default value is in use if not yet set.

But I already removed transformation of configs older than `CONFIG_VERSION = 4` in a983fa67f4c58a2e1483e874658211b468dcd470. So nothing to do!?
